src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -941,7 +934,6 @@ def get_create_table_as_select_plan(child: SnowflakePlan, replace, error):
                 child,
                 source_plan,
                 is_ddl_on_temp_object=is_temp_table_type,
-                propagate_referenced_ctes=False,
```

Comment:
let's actually move that into a separate pr, and add more tests in test query generator with manually constructed plan tree to make sure the cte is propogated and used corretly